src/table-procedure/src/create.rs

```diff
@@ -132,23 +132,22 @@ impl CreateTableProcedure {
     }
 
     async fn on_prepare(&mut self) -> Result<Status> {
-        if !self
+        let table_exists = self
             .catalog_manager
-            .schema_exist(
+            .table_exist(
                 &self.data.request.catalog_name,
                 &self.data.request.schema_name,
+                &self.data.request.table_name,
             )
             .await
-            .context(AccessCatalogSnafu)?
-        {
-            logging::error!(
-                "Failed to create table {}, schema not found",
-                self.data.table_ref(),
-            );
-            return SchemaNotFoundSnafu {
-                name: &self.data.request.schema_name,
+            .context(AccessCatalogSnafu)?;
+        if table_exists && !self.data.request.create_if_not_exists {
+            return TableNotFoundSnafu {
```

Comment:
Should it be `TableAlreadyExist` instead?

Reply:
Oops 😢